modules/core/src/main/java/org/apache/ignite/configuration/ClientConfiguration.java

```diff
@@ -33,9 +36,16 @@
     /** Serial version uid. */
     private static final long serialVersionUID = 0L;
 
-    /** @serial Server addresses. */
+    /**
+     * @serial Server addresses.
+     * @deprecated deprecated in favor of {@link #addrFinder}. Keep it due to Serializable compatibility.
+     */
+    @Deprecated
     private String[] addrs = null;
 
+    /** Server addresses finder. */
+    private transient Supplier<String[]> addrFinder;
```

Comment:
I would prefer a dedicated interface - `ThinClientAddressFinder` or something like this.

Reply:
Used `ClientAddressFinder` as all classes in this package use Client prefix.